pkg/webhooks/provisioning/v1alpha1/defaulter.go

```diff
@@ -49,20 +53,39 @@ func (v *Defaulter) Handle(ctx context.Context, req admission.Request) admission
 	if err != nil {
 		return admission.Errored(http.StatusBadRequest, err)
 	}
-	v.applyDefaults(&provisioner.Spec)
+	if err := v.applyDefaults(ctx, &provisioner.Spec); err != nil {
+		return admission.Errored(0, fmt.Errorf("applying defaults to provisioner, %w", err))
+	}
+
 	marshaled, err := json.Marshal(provisioner)
 	if err != nil {
 		return admission.Errored(http.StatusInternalServerError, err)
 	}
 	return admission.PatchResponseFromRaw(req.Object.Raw, marshaled)
 }
 
-func (v *Defaulter) applyDefaults(spec *provisioning.ProvisionerSpec) {
-	v.defaultTTL(spec)
+func (v *Defaulter) applyDefaults(ctx context.Context, spec *provisioning.ProvisionerSpec) error {
+	return functional.ValidateAll(
+		func() error { return v.defaultTTL(spec) },
+		func() error { return v.defaultCapacityType(ctx, spec) },
+	)
 }
 
-func (v *Defaulter) defaultTTL(spec *provisioning.ProvisionerSpec) {
+func (v *Defaulter) defaultTTL(spec *provisioning.ProvisionerSpec) error {
 	if spec.TTLSeconds == nil {
 		spec.TTLSeconds = ptr.Int32(300)
 	}
+	return nil
+}
+
+func (v *Defaulter) defaultCapacityType(ctx context.Context, spec *provisioning.ProvisionerSpec) error {
```

Comment:
Would love to figure out how to make these defaulters not errorable.

Reply:
I think it's fair to just remove the error. I had it this way in case the cloud provider wanted to dynamically retrieve the possible capacity types.  I don't think they'll change very often... unless we would put placement groups into the capacityType umbrella...